src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -534,7 +505,7 @@ public void RelayTransaction(uint256 hash)
         {
             this.logger.LogTrace("()");
 
-            if (!this.CanSend)
+            if (!this.AttachedPeer?.PeerVersion?.Relay ?? true)  
```

Comment:
Instead just don't select this peer for sending the Trickle:

```
                var behaviours = peers.Select(s => s.Behavior<MempoolBehavior>());
                foreach (var behaviour in behaviours)
                    await behaviour.SendTrickleAsync().ConfigureAwait(false);
```

in MempoolSignaled.


do there:

```
var behaviours = peers.Where(x => x.PeerVersion != null ? x.PeerVersion.Relay : true).Select(x => x.Behavior<MempoolBehavior>());
```